server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -396,7 +397,14 @@ private TableMgmtStats manageTablets(Iterator<TabletManagement> iter,
       }
 
       if (tm.getOperationId() != null) {
-        goal = TabletGoalState.UNASSIGNED;
+        // If there are still wals the tablet needs to be hosted
+        // to process the wals before starting the merge op
+        if (tm.getLogs().isEmpty()
+            && tm.getOperationId().getType() == TabletOperationType.MERGING) {
+          goal = TabletGoalState.UNASSIGNED;
+        } else {
+          goal = TabletGoalState.HOSTED;
+        }
```

Comment:
When a tablet has an operationid, generally want it unassigned.  So need to work in an exception of that in the case of walogs and merge

```suggestion
        if (!tm.getLogs().isEmpty()
            && tm.getOperationId().getType() == TabletOperationType.MERGING) {
          goal = TabletGoalState.HOSTED;
        } else {
          goal = TabletGoalState.UNASSIGNED;
        }
```